src/launcher/java/org/truffleruby/launcher/RubyLauncher.java

```diff
@@ -13,27 +13,88 @@
 import org.graalvm.launcher.AbstractLanguageLauncher;
 import org.graalvm.options.OptionCategory;
 import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Engine;
+import org.graalvm.polyglot.PolyglotException;
+import org.graalvm.polyglot.Source;
+import org.truffleruby.launcher.options.CommandLineException;
 import org.truffleruby.launcher.options.CommandLineOptions;
+import org.truffleruby.launcher.options.CommandLineParser;
+import org.truffleruby.launcher.options.ExecutionAction;
 import org.truffleruby.launcher.options.OptionsCatalog;
 
+import java.io.File;
+import java.io.PrintStream;
+import java.lang.reflect.Method;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
 public class RubyLauncher extends AbstractLanguageLauncher {
 
+    public static final String LANGUAGE_ID = "ruby";
+    public static final String ENGINE_ID = "truffleruby";
+    public static final String LANGUAGE_VERSION = "2.3.6";
+    public static final String LANGUAGE_BASE_VERSION = "2.3.0";
+    public static final int LANGUAGE_REVISION = 61260;
+    public static final String BOOT_SOURCE_NAME = "main_boot_source";
+    public static final String RUBY_COPYRIGHT = "truffleruby - Copyright (c) 2013-2018 Oracle and/or its affiliates";
+
+    // Properties set directly on the java command-line with -D for image building
+    public static final String LIBSULONG_DIR = isAOT() ? System.getProperty("truffleruby.native.libsulong_dir") : null;
+    public static final boolean PRE_INITIALIZE_CONTEXTS = System.getProperty("polyglot.engine.PreinitializeContexts") != null;
+
     private final CommandLineOptions config = new CommandLineOptions();
 
     public static void main(String[] args) {
         new RubyLauncher().launch(args);
     }
 
+    static boolean isGraal() {
+        return Engine.create().getImplementationName().contains("Graal");
```

Comment:
Should the Engine be closed or it doesn't matter?
Ideally it'd be a static method so that wouldn't be a concern of course.

Reply:
It's fine, from javadoc: "It can be useful to `.create()` an engine instance without a context to only access meta-data for installed languages, instruments and their available options."

Reply:
Please close any engine you create after use.

Reply:
@pitr-ch Could you make a PR to close() that temporary Engine?

Reply:
Already done.